crates/bevy_gltf/src/loader.rs

```diff
@@ -291,36 +289,174 @@ async fn load_gltf<'a, 'b, 'c>(
                     return Err(GltfError::MissingAnimationSampler(animation.index()));
                 };
 
-                let keyframes = if let Some(outputs) = reader.read_outputs() {
+                if keyframe_timestamps.is_empty() {
+                    warn!("Tried to load animation with no keyframe timestamps");
+                    continue;
+                }
+
+                let maybe_curve: Option<VariableCurve> = if let Some(outputs) =
+                    reader.read_outputs()
+                {
                     match outputs {
                         ReadOutputs::Translations(tr) => {
-                            Keyframes::Translation(tr.map(Vec3::from).collect())
+                            let translations: Vec<Vec3> = tr.map(Vec3::from).collect();
+                            if keyframe_timestamps.len() == 1 {
+                                Some(VariableCurve::Translation(TranslationCurve::Constant(
+                                    constant_curve(Interval::EVERYWHERE, translations[0]),
+                                )))
+                            } else {
+                                match interpolation {
+                                    gltf::animation::Interpolation::Linear => {
+                                        UnevenSampleAutoCurve::new(
+                                            keyframe_timestamps.into_iter().zip(translations),
+                                        )
+                                        .ok()
+                                        .map(|c| {
+                                            VariableCurve::Translation(TranslationCurve::Linear(c))
+                                        })
```

Comment:
I'm wondering why clippy isn't complaining but (nit-picky and opinionated!!) I think this can be

```suggestion
                                       .map(TranslationCurve::Linear)
                                       .map(VariableCurve::Translation)
```

saving us the indentation, a line of code and generally easier read (probably just to me). This is basically applicaable everywhere below